components/camera/camera_test.go

```diff
@@ -250,3 +251,172 @@ func TestCameraWithProjector(t *testing.T) {
 
 	test.That(t, cam2.Close(context.Background()), test.ShouldBeNil)
 }
+
+// testCamera is a simple camera implementation for testing GetImage and GetImages utility functions.
+type testCamera struct {
+	resource.Named
+	imageFunc  func(ctx context.Context, mimeType string, extra map[string]interface{}) ([]byte, camera.ImageMetadata, error)
+	imagesFunc func(ctx context.Context) ([]camera.NamedImage, resource.ResponseMetadata, error)
+}
+
+func (c *testCamera) Image(ctx context.Context, mimeType string, extra map[string]interface{}) ([]byte, camera.ImageMetadata, error) {
+	return c.imageFunc(ctx, mimeType, extra)
+}
+
+func (c *testCamera) Images(ctx context.Context) ([]camera.NamedImage, resource.ResponseMetadata, error) {
+	if c.imagesFunc != nil {
+		return c.imagesFunc(ctx)
+	}
+	return nil, resource.ResponseMetadata{}, errors.New("not implemented")
+}
+
+func (c *testCamera) NextPointCloud(ctx context.Context) (pointcloud.PointCloud, error) {
+	return nil, errors.New("not implemented")
+}
+
+func (c *testCamera) Properties(ctx context.Context) (camera.Properties, error) {
+	return camera.Properties{}, nil
+}
+
+func (c *testCamera) Close(ctx context.Context) error {
+	return nil
+}
+
+func (c *testCamera) Reconfigure(ctx context.Context, deps resource.Dependencies, conf resource.Config) error {
+	return nil
+}
+
+// verifyImageEquality compares two images and verifies they are identical.
+func verifyImageEquality(t *testing.T, img1, img2 image.Image) {
+	t.Helper()
+	diff, _, err := rimage.CompareImages(img1, img2)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, diff, test.ShouldEqual, 0)
+}
+
+// verifyDecodedImage verifies that decoded image bytes match the original image.
+func verifyDecodedImage(t *testing.T, imgBytes []byte, mimeType string, originalImg image.Image) {
+	t.Helper()
+	test.That(t, len(imgBytes), test.ShouldBeGreaterThan, 0)
+
+	// For JPEG, compare the raw bytes after encoding the original image
+	if mimeType == rutils.MimeTypeJPEG {
+		expectedBytes, err := rimage.EncodeImage(context.Background(), originalImg, mimeType)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, imgBytes, test.ShouldResemble, expectedBytes)
+		return
+	}
```

Comment:
Shouldn't we define bytes tests for all mime types or fail with an error if the test encounters an unexpected mime type?

Reply:
I was originally only testing for image.Image equality and put in the bytes test for jpeg only because jpeg is lossy and `rimage.CompareImages` calls fail for them. Could you elaborate on why you feel bytes comparisons is better suited for this test than the `rimage.CompareImages` utility?

Reply:
Added comment